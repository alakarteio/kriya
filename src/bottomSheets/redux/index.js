import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
