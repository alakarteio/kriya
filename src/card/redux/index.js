export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
