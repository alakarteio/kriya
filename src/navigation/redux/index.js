export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
