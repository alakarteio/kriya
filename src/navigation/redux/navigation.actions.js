export const ADD_STEP = 'NAVIGATION_ADD_STEP'
export const addStep = step => ({ type: ADD_STEP, payload: step })

export const RESET_STEPS = 'NAVIGATION_RESET_STEPS'
export const resetSteps = () => ({ type: RESET_STEPS })

export const CLICKED = 'NAVIGATION_CLICKED'
export const click = number => ({ type: CLICKED, payload: number })
