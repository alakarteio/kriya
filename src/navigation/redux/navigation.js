import { ADD_STEP, RESET_STEPS } from './navigation.actions'

export const initState = []
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case ADD_STEP: return [...state, payload]
    case RESET_STEPS: return initState
    default: return state
  }
}
