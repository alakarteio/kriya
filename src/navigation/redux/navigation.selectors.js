export const getNavigation = state => state.ui.navigation
