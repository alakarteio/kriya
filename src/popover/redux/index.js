export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
