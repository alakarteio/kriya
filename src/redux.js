export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
