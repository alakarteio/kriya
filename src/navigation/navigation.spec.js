/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import Navigation from './navigation'

const snap = (props) => {
  const store = createStore(() => ({}))
  /* eslint-disable react/no-children-prop */
  const component = renderer.create(
    <Provider store={store}>
      <Navigation children={[]} {...props} />
    </Provider>,
  )
  /* eslint-enable react/no-children-prop */

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Navigation', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should have a default behaviour', () => snap({}))
    it('should handle one step', () => snap({ children: ['step 1'] }))
    it('should handle multiple steps', () => snap({ children: ['step 1', 'step 2'] }))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
