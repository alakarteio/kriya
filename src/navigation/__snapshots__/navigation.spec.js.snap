// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Navigation graphical (JSX) should add custom className 1`] = `
<div
  className="navigation custom"
  style={Object {}}
/>
`;

exports[`common/Navigation graphical (JSX) should add custom style 1`] = `
<div
  className="navigation"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;

exports[`common/Navigation graphical (JSX) should handle multiple steps 1`] = `
<div
  className="navigation"
  style={Object {}}
>
  <div
    className="step intermediate"
    style={Object {}}
  >
    <button
      onClick={[Function]}
    >
      step 1
    </button>
    <div
      className="separator"
    >
       /
    </div>
  </div>
  <div
    className="step last"
    style={Object {}}
  >
    <button
      onClick={undefined}
    >
      step 2
    </button>
  </div>
</div>
`;

exports[`common/Navigation graphical (JSX) should handle one step 1`] = `
<div
  className="navigation"
  style={Object {}}
>
  <div
    className="step last"
    style={Object {}}
  >
    <button
      onClick={undefined}
    >
      step 1
    </button>
  </div>
</div>
`;

exports[`common/Navigation graphical (JSX) should have a default behaviour 1`] = `
<div
  className="navigation"
  style={Object {}}
/>
`;
